flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/GenericHiveMetastoreCatalog.java

```diff
@@ -95,144 +106,103 @@ public void alterDatabase(String name, CatalogDatabase newDatabase, boolean igno
 	// ------ tables and views------
 
 	@Override
-	public void dropTable(ObjectPath tablePath, boolean ignoreIfNotExists)
-			throws TableNotExistException, CatalogException {
-		try {
-			client.dropTable(
-				tablePath.getDatabaseName(),
-				tablePath.getObjectName(),
-				// Indicate whether associated data should be deleted.
-				// Set to 'true' for now because Flink tables shouldn't have data in Hive. Can be changed later if necessary
-				true,
-				ignoreIfNotExists);
-		} catch (NoSuchObjectException e) {
-			if (!ignoreIfNotExists) {
-				throw new TableNotExistException(catalogName, tablePath);
-			}
-		} catch (TException e) {
-			throw new CatalogException(
-				String.format("Failed to drop table %s", tablePath.getFullName()), e);
+	public void validateCatalogBaseTable(CatalogBaseTable table) throws IllegalArgumentException {
+		// TODO: invalidate HiveCatalogView
+		if (table instanceof HiveCatalogTable) {
```

Comment:
I'm wondering if we should specify what we are expecting and throw exceptions if condition isn't met.